integration/ingester_sharding_test.go

```diff
@@ -109,6 +103,12 @@ func TestIngesterSharding(t *testing.T) {
 				}
 			}
 
+			// Verify that the expected number of ingesters had series (write path). However,
+			// we _don't_ verify that a subset of ingesters were queried for the series (read
+			// path). This is because the way which ingesters to query is calculated depends on
+			// when they were registered compared to the "query ingesters within" time. They'll
+			// never be registered long enough as part of this test to be queried for the series.
+			// Instead, all ingesters are queried.
```

Comment:
Yeah, this is not very good. We're not really testing shuffle sharding anymore here. What if we set a query-ingesters-within to a low value (e.g. 3s), then we start ingesters, we wait 3s, we push samples and query back.

Reply:
Yes! I tried out using a small value (30s) and a `time.Sleep()` locally. I was able to get the tests passing once but they seemed a little flaky. I'll look into this more and see if I can get them working reliably.

Reply:
Make sure to set very very low ring heartbeat period, otherwise the registration timestamp is not updated frequently enough.